### PR TITLE
Window with bars will be frame with bars after hitting

### LINKED ISSUE #13773: Corrosive zombie's acid spit fly through windows with bars
default
0.C-3790, Win 10, tiles.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/27480237-corrosive-zombie-s-acid-spit-fly-through-windows-with-bars?utm_campaign=plugin&utm_content=tracker%2F146201&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F146201&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Night-Pryanik**
Fixes #13773.

### CODE REVIEW

File: src/map.cpp [hunk 1]
```diff
@@ -3873,6 +3873,12 @@ void map::shoot( const tripoint &p, projectile &proj, const bool hit_items )
                 ter_set(p, t_window_frame);
             }
         }
+    } else if( terrain == t_window_bars_alarm ) {
+        dam -= rng(1,3);
+        if (dam > 0) {
+                sounds::sound(p, 16, _("glass breaking!"), false, "smash", "glass");
+            ter_set(p, t_window_bars);
```
Comment (BevapDin) [inline src/map.cpp]: Have you tested this? It shouldn't work as the variable `t_window_bars` has not been initialized anywhere. Consider removing the changes in mapdata.h and mapdata.cpp and using a string literal here: `ter_set( p, "t_window_bars" )`

**Comment by Coolthulhu** [comment]
Threw a stone through the window with bars, it caused an alarm and turned into nothing.

**Comment by Night-Pryanik** [comment]
@Coolthulhu @BevapDin tested right now. 
Intact window with bars:

Threw a rock through the window:

Got alarm, window glass is broken, window with bars transformed into window frame with bars.

**Comment by BevapDin** [comment]
So where exactly is the variable `t_window_bars` initialized? Currently it's only declared and defined.

**Comment by Night-Pryanik** [comment]
I'm not sure what "initialized" means.
I used this piece of code as example from just a few lines higher:
```
else if( terrain == t_window_boarded ) {
        dam -= rng(10, 30);
        if (dam > 0) {
                sounds::sound(p, 16, _("glass breaking!"),  false, "smash", "glass");
            ter_set(p, t_window_frame);
```
I though that `t_window_frame` isn't initialized anywhere in map.cpp, but I found it in mapdata.cpp and mapdata.h. So I added my `t_window_bars` there too and compiler stopped complaining about not finding `t_window_bars`.

**Comment by BevapDin** [comment]
> I'm not sure what "initialized" means.

Initialized means it has been set to a specific initial value.

You have declared a new variable In mapdata.h.
You have defined the variable in mapdata.cpp.

But you have not assigned a specific value to it. It's value is 0 (being a static global variable). It's certainly not the correct value (which can only be know at runtime at is most likely not 0).

For example, the variable `t_window_frame` is initialized to the correct value at mapdata.cpp line 684: `t_window_frame=terfind("t_window_frame");`

**Comment by Night-Pryanik** [comment]
I'm terrible, terrible sorry. I did all the initialization in my C::B build and it worked all right, but forgot to initialize it in PR.

**Comment by Coolthulhu** [comment]
Could also drop glass - currently it only loses the glass part of the window.

**Comment by Coolthulhu** [comment]
Also, acetylene torch can't cut through those.

**Comment by chaosvolt** [comment]
Ooh, I missed that. o3o

**Pull request merged**